### PR TITLE
refactor: split input component into typed input

### PR DESCRIPTION
**Pull request by hlavkat**

### CODE REVIEW

File: src/lib/components/ui/input/number-input.svelte [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<script lang="ts">
+	import { Add, Subtract } from 'carbon-icons-svelte'
+	import Button from '../button.svelte'
+	import Input, { type Props } from './input.svelte'
+	import type { HTMLInputAttributes } from 'svelte/elements'
+	let {
+		value = $bindable(0),
```
Comment (vojtechsimetka) [inline src/lib/components/ui/input/number-input.svelte:7]: Maybe here the type of value could be hardcoded to `number`

File: src/lib/components/ui/input/search-input.svelte [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{/snippet}
+
+<Input
+	type="search"
+	{dimension}
+	{disabled}
+	iconStart={!withButton ? iconStart : undefined}
+	{buttons}
+	bind:value
+	{...restProps}
```
Comment (vojtechsimetka) [inline src/lib/components/ui/input/search-input.svelte:61]:
```suggestion
<Input
	{dimension}
	{disabled}
	iconStart={!withButton ? iconStart : undefined}
	{buttons}
	bind:value
	{...restProps}
	type="search"
```

File: src/lib/components/ui/input/search-input.svelte [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	interface SearchProps extends Props {
+		withButton?: boolean
+	}
+	let {
+		value = $bindable(),
+		dimension,
+		disabled,
+		withButton = false,
+		...restProps
+	}: SearchProps & HTMLInputAttributes = $props()
```
Comment (vojtechsimetka) [inline src/lib/components/ui/input/search-input.svelte:15]:
```suggestion
	}: SearchProps & Omit<HTMLInputAttributes, 'type'> = $props()
```

**Review by vojtechsimetka — APPROVED** [review]

**Review by agazso — APPROVED** [review]
Looks good.

The only issue I found is that the Date input still has the default calendar icon, but this can be addressed in a separate PR.

**Pull request merged**